conformance/lang/expressions/xml_template_expr.balt

```diff
@@ -0,0 +1,765 @@
+    io:println(text); // @output James
+}
+
+Test-Case: output
+Description: Test usage of let expressions as interpolated expressions in xml content.
+Fail-Issue: ballerina-platform/ballerina-lang#33077
+Labels: xml-template-expr, xml, let-expr
+
+public function init() {
+    xml x = xml `<Books>${xml `<Book>"${let int y = 5 in y}"</Book>`}</Books>`;
+    io:println(x); // @output <Books><Book>"5"</Book></Books>
+
+    xml<xml:Comment> c = xml `<!--${let string name = "James" in name}-->`;
+    io:println(c); // @output <!--James-->
+
+    xml z = xml `<Books>${xml `<Book author="${let string name = "James" in name}">"Ballerina"</Book>`}</Books>`;
+    io:println(z); // @output <Books><Book author="James">"Ballerina"</Book></Books>
+
+    xml<xml:Text> text = xml `${let string name = "James" in name}`;
+    io:println(text); // @output James
+}
+
+Test-Case: output
+Description: Test usage of field access expressions as interpolated expressions in xml content.
+Labels: xml-template-expr, xml, field-access-expr
+
+type Book record {
+    string name;
+    float price;
+    string author;
+};
+
+public function init() {
+    Book book = {name : "Ballerina", price: 20.0, author: "James"};
+
+    xml x = xml `<Books>${xml `<Book>"${book.name}"</Book>`}</Books>`;
+    io:println(x); // @output <Books><Book>"Ballerina"</Book></Books>
+
+    xml<xml:Comment> y = xml `<!--${book.author}-->`;
+    io:println(y); // @output <!--James-->
+
+    xml z = xml `<Books>${xml `<Book author="${book.author}">"${book.name}"</Book>`}</Books>`;
+    io:println(z); // @output <Books><Book author="James">"Ballerina"</Book></Books>
+
+    xml<xml:Text> text = xml `${book.author}`;
+    io:println(text); // @output James
+}
+
+Test-Case: error
+Description: Test usage of optional field access expressions as interpolated expressions in xml content.
+Labels: xml-template-expr, xml, optional-field-access-expr
+
+type Book record {
+    string name;
+    float price;
+    string author;
+    string coAuthor?;
+};
+
+public function init() {
+    Book book = {name : "Ballerina", price: 20.0, author: "James", coAuthor: "Sanjiva"};
+
+    xml x = xml `<Books>${xml `<Book>"${book?.coAuthor}"</Book>`}</Books>`; // @error static type of the interpolated expressions in xml content should not be nil
+    xml<xml:Comment> y = xml `<!--${book?.coAuthor}-->`; // @error static type of the interpolated expressions in comments should not be nil
+    xml z = xml `<Books>${xml `<Book author="${book?.coAuthor}">"${book.name}"</Book>`}</Books>`; // @error static type of the expressions in attribute values should not be nil
+    xml<xml:Text> text = xml `${book?.coAuthor}`; // @error static type of the interpolated expressions in xml text should not be nil
+}
+
+Test-Case: error
+Description: Test usage of member access expressions as interpolated expressions in xml content.
+Labels: xml-template-expr, xml, member-access-expr
```

Comment:
Shall we add the missing labels here and please check other places in the file as well, there seems to be quite a bit that has been missed.